### ISSUE TITLE
Should the project re-iterate to become a cross-platform application

### ISSUE THREAD

**Issue by Venkat5694** [abahmed/Deer#19]
As per the README.md, Deer aims to be cross-platform and to achieve that should this project take a different direction. May be pivot to an [Electron](https://github.com/electron/electron) based application so that it can become a cross-platform application.

@abahmed Thoughts and comments?

**Comment by ai-zahran** [comment]
We are actually intending to experiment with this approach. Maybe on a standalone branch.
When starting, the direction we intended to go through was to make it easier for students who mostly know C++ to contribute.
But through time, we became anxious about the slow process of growing and the problems of cross-platform integration to come if we do not pivot to a cross-platform framework, which made us consider frameworks like Electron.

Would you like to contribute with us though this direction?

**Comment by Venkat5694** [comment]
Sure. Let me know once you have a project created under this repo and I can start pitching in.

**Comment by abahmed** [comment]
Thanks @Venkat5694  for your suggestion. Initially, we started this project to help new comers to open source involve, and we used C++ to implement Deer for the reasons that @ai-zahran mentioned.

As you said, it's needed to re-iterate the project to support each platform, but this will be hard to maintain. Also, adding more features, simply means increase complexity of Deer as C++ does not have flexible utilities like JS.

BTW, I agree with your suggestion and I'll create a branch for re-implementing Deer using Electron and it will be merged to master branch since it reaches current state of C++ implementation.
